### PR TITLE
IEP 325 save git and python executable paths

### PR DESCRIPTION
**Pull request by sigmaaa**
Git and python paths are now storing in the eclipse preferences. These preferences are updating on the install tools button click.
Covered test cases:
- Test case 1: go to Help -> ESP-IDF tools manager -> Install Tools. Choose git and python paths and click Install Tools. After restarting eclipse and opening Install Tools again there will be chosen paths.
- Test case 2: Choose new git and python paths, but don't click on install tools. Close the Install tools window and open it again. There will be old git and python paths.
- Test case 3: Preferences are stored in <workspace>/.metadata/.plugins/org.eclipse.core.runtime/.settings/com.espressif.idf.ui.preference.prefs. Go to this file and delete preferences. Then open eclipse again and go to Install Tools window, there will be default locations for git and python.

### CODE REVIEW

**Review by kolipakakondal — APPROVED** [review]
LGTM

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/DirectorySelectionDialog.java [hunk 1]
```diff
@@ -47,17 +50,20 @@
 	private Text gitLocationtext;
 	private Text pythonLocationtext;
 	private String commandId;
-
+	private static final String executablePreferencesNode = "com.espressif.idf.ui.preference";
+	private static final String pythonPathNodeKey = "PYTHON_EXECUTABLE";
+	private static final String gitPathNodeKey = "GIT_EXECUTABLE";
```
[the remark this answers is not in the document]
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/DirectorySelectionDialog.java]: @kolipakakondal non-externalized string literal was added
https://github.com/espressif/idf-eclipse-plugin/blob/1b076750f95c702db3ea9a31fc737081c1941c89/bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/DirectorySelectionDialog.java#L53-L54

UI strings was externalize:
https://github.com/espressif/idf-eclipse-plugin/blob/1b076750f95c702db3ea9a31fc737081c1941c89/bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/messages.properties#L15-L18

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/DirectorySelectionDialog.java [hunk 2]
```diff
@@ -47,17 +50,20 @@
 	private Text gitLocationtext;
 	private Text pythonLocationtext;
 	private String commandId;
-
+	private static final String executablePreferencesNode = "com.espressif.idf.ui.preference";
```
[the remark this answers is not in the document]
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/DirectorySelectionDialog.java]: @kolipakakondal 

Good point, I added this method:

https://github.com/espressif/idf-eclipse-plugin/blob/1b076750f95c702db3ea9a31fc737081c1941c89/bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/DirectorySelectionDialog.java#L312-L315

And then I changed methods which where using this property to these: 
https://github.com/espressif/idf-eclipse-plugin/blob/1b076750f95c702db3ea9a31fc737081c1941c89/bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/DirectorySelectionDialog.java#L252-L260

**Comment by kolipakakondal** [comment]
@sigmaaa Could you resolve these conflicts by rebasing with the latest changes?

**Comment by kolipakakondal** [comment]
Changes LGTM

**Review by kolipakakondal — APPROVED** [review]
LGTM

**Pull request merged**